RealmSwift/Migration.swift

```diff
@@ -0,0 +1,233 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2014 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Realm
+import Realm.Private
+
+/**
+Migration block used to migrate a Realm.
+
+:param: migration        `Migration` object used to perform the migration. The
+                         migration object allows you to enumerate and alter any
+                         existing objects which require migration.
+:param: oldSchemaVersion The schema version of the `Realm` being migrated.
+*/
+public typealias MigrationBlock = (migration: Migration, oldSchemaVersion: UInt) -> Void
+
+/// Object class used during migrations
+public typealias MigrationObject = DynamicObject
+
+/**
+Provides both the old and new versions of an object in this Realm. Objects properties can only be
+accessed using subscripting.
+
+:param: oldObject Object in original `Realm` (read-only)
+:param: newObject Object in migrated `Realm` (read-write)
+*/
+public typealias MigrationObjectEnumerateBlock = (oldObject: MigrationObject, newObject: MigrationObject) -> Void
```

Comment:
Yeah, after looking at the obj-c code there's some code paths that explicitly pass nil.